fluss-common/src/main/java/org/apache/fluss/row/arrow/writers/ArrowMapWriter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.fluss.row.arrow.writers;
+
+import org.apache.fluss.row.DataGetters;
+import org.apache.fluss.row.InternalArray;
+import org.apache.fluss.row.InternalMap;
+import org.apache.fluss.shaded.arrow.org.apache.arrow.vector.FieldVector;
+import org.apache.fluss.shaded.arrow.org.apache.arrow.vector.complex.MapVector;
+import org.apache.fluss.shaded.arrow.org.apache.arrow.vector.complex.StructVector;
+
+/** {@link ArrowFieldWriter} for Map. */
+public class ArrowMapWriter extends ArrowFieldWriter {
+
+    private final ArrowFieldWriter keyWriter;
+    private final ArrowFieldWriter valueWriter;
+    private int offset;
+
+    public ArrowMapWriter(
+            FieldVector fieldVector, ArrowFieldWriter keyWriter, ArrowFieldWriter valueWriter) {
+        super(fieldVector);
+        this.keyWriter = keyWriter;
+        this.valueWriter = valueWriter;
+    }
+
+    @Override
+    public void doWrite(int rowIndex, DataGetters row, int ordinal, boolean handleSafe) {
+        InternalMap map = row.getMap(ordinal);
+        MapVector mapVector = (MapVector) fieldVector;
+        StructVector structVector = (StructVector) mapVector.getDataVector();
+
+        InternalArray keyArray = map.keyArray();
+        InternalArray valueArray = map.valueArray();
+
+        mapVector.startNewValue(rowIndex);
+        for (int i = 0; i < map.size(); i++) {
+            int fieldIndex = offset + i;
+            structVector.setIndexDefined(fieldIndex);
+            keyWriter.write(fieldIndex, keyArray, i, handleSafe);
+            valueWriter.write(fieldIndex, valueArray, i, handleSafe);
+        }
+        offset += map.size();
```

Comment:
we should override `reset()` method to reset `offset` and `keyWriter` and `valueWriter`. 

I also found that we forgot to add the `reset()` for the newly introduced `ArrowRowWriter`.